docs/userguide/glossary.rst

```diff
@@ -0,0 +1,218 @@
+
+A manager in Parsl is responsible for overseeing the execution of tasks on specific compute resources. It's like a supervisor who ensures that all workers (or workers within a block) are carrying out their tasks correctly and efficiently.
+
+.. _memoizationglossary:
+
+**Memoization:**
+-------------------
+
+Memoization is like remembering something so you don't have to do it again. In Parsl, it means that if you run an app with the same inputs multiple times, Parsl will remember the result from the first time and give it to you again instead of running the app again. This can save a lot of time.
+
+.. _mpiappglossary:    
+
+**MPI App:**
+---------------
+
+An MPI app is a special kind of app that uses a technology called Message Passing Interface (MPI) to communicate between different computers. It's like a walkie-talkie that lets different apps talk to each other.
+
+.. _nodeglossary:
+
+**Node:**
+------------
+
+A node in Parsl is like a workstation in a factory. It's the physical or virtual machine where work gets done. Each node provides the computational power needed to run tasks, and it can host several workers who carry out the tasks.
+
+.. _parallelismglossary:
+
+**Parallelism:**
+-------------------
+
+Parallelism means doing multiple things at the same time. In Parsl, it means that your apps can run simultaneously on different computers. This can make your programs run much faster.
+
+.. _parslscriptglossary:    
+
+**Parsl Script:**
+---------------------
+
+A Parsl script is a file that contains the instructions for how to run your apps in parallel. It's like a recipe that tells you what ingredients to use and how to combine them.
+
+.. _pluginglossary:
+
+**Plugin:**
+---------------
+
+A plugin is like an add-on for Parsl. It's a piece of code that you can add to Parsl to give it new features or change how it works. It's like an extra tool that you can add to your toolbox.
+
+.. _pythonappglossary: 
+
+**Python App:**
+------------------
+
+A Python app is a special kind of app in Parsl that's written in the Python programming language. It's a way to use Parsl to run your Python code in parallel.
+
+.. _resourceglossary:
+
+**Resource:**
+---------------
+
+A resource in Parsl refers to any computational asset that can be used to execute tasks, such as CPU cores, memory, or entire nodes. It's like the tools and materials you need to get a job done.
+
+.. _serializationglossary:    
+
+**Serialization:**
+--------------------
+
+Serialization is like packing your belongings into a suitcase so you can take them on a trip. In Parsl, it means converting your data into a format that can be sent over a network to another computer.
+
+.. _stagingglossary:    
+
+**Staging:**
+---------------
+
+Staging is like setting the stage for a play. In Parsl, it means preparing the data that your apps need before they start running. This can involve things like copying files to the right location or converting them into the right format.
```

Comment:
I don't think of staging as preparing the data or converting them, but just of moving them. But I'm not 100% this is correct for Parsl...